proxy-node-gateway/src/main/java/uk/gov/ida/notification/GatewayApplication.java

```diff
@@ -120,6 +120,15 @@ private void setRequestServletFilter(Environment environment) {
     }
 
     private void setResponseServletFilter(Environment environment) {
+
```

Comment:
Registering this filter first is deliberate; it ensures it's invoked last after filter chain processing.

Reply:
Is there a more robust way of enforcing that this filter always gets invoked last in the chain?

Reply:
And does the order of filters even matter? And if it does - should it?

Reply:
The order is important in that any other filter that accesses the session would have to run beforehand.
I don't think there's another way to register servlet filter order. If we used a ContainerResponseFilter, the execution order would still have to be declared.